### PR TITLE
#34895 ophalen taken op basis van ingelogde gebruiker

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

**Review by theo-ritense — APPROVED** [review]

File: gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/client/ObjectsApiClient.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            .get()
+            .uri { uriBuilder ->
+                uriBuilder.path("/api/v2/objects").queryParam("page", page)
+                objectTypeUrl?.let { uriBuilder.queryParam("type", it) }
+                uriBuilder.queryParam("data_attrs", ObjectSearchParameter.toQueryParameter(objectSearchParameters))
+                uriBuilder.queryParam("ordering", ordering)
+                uriBuilder.build()
+            }
+            .retrieve()
+            .awaitBody<ResultPage<ObjectsApiObject>>()
```
Comment (ivo-ritense) [inline gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/client/ObjectsApiClient.kt:48]: Did you consider using generics for this?

File: gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/service/TaskService.kt [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            is BurgerAuthentication -> ObjectSearchParameter("bsn", "exact", authentication.getBsn())
+            is BedrijfAuthentication -> ObjectSearchParameter("kvk", "exact", authentication.getKvkNummer())
+            else -> throw UserTypeUnsupportedException("User type not supported")
+        }
+        val statusOpenSearchParameter = ObjectSearchParameter("status", "exact", "open")
+
+        return objectsApiClient.getObjects(
+            objectSearchParameters = listOf(userSearchParameter, statusOpenSearchParameter),
+            objectTypeUrl = objectsApiTaskConfig.typeUrl,
+            page = 1,
```
Comment (ivo-ritense) [inline gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/service/TaskService.kt:43]: Should we not support pagination?

File: gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/domain/ObjectSearchParameter.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ritense.valtimo.portal.gzac.objectsapi.domain
+
+data class ObjectSearchParameter(
+    val attribute: String,
+    val comparator: String,
```
Comment (ivo-ritense) [inline gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/domain/ObjectSearchParameter.kt:20]: We could use an enum for this.

Comment (Klaas-Ritense) [reply gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/domain/ObjectSearchParameter.kt]: Added

File: gzac/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * limitations under the License.
+ */
+plugins {
+    kotlin("jvm")
+}
+
+val isLib = true
+
+dependencies {
+    implementation(kotlin("stdlib-jdk8"))
```
[the remark this answers is not in the document]
Comment (Klaas-Ritense) [reply gzac/build.gradle.kts]: removed

File: gzac/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+plugins {
+    kotlin("jvm")
+}
+
+val isLib = true
```
[the remark this answers is not in the document]
Comment (Klaas-Ritense) [reply gzac/build.gradle.kts]: removed

File: gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/graphql/TaskPage.kt [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+data class TaskPage(
+    val results: List<Task>,
+    val nextPage: Int?
+) {
+    companion object {
+        fun fromResultPage(resultPage: ResultPage<ObjectsApiObject>): TaskPage {
+            return TaskPage(
+                results = resultPage.results.map { Task.fromObjectsApiObject(it) },
+                nextPage = resultPage.getNextPageNumber()
```
Comment (ivo-ritense) [inline gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/task/graphql/TaskPage.kt:30]: Looks good, but i wonder if we need to show the total number of results?

File: gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/client/ObjectsApiClient.kt (hunk 1 above)
Comment (theo-ritense) [reply gzac/objects-api/src/main/kotlin/com/ritense/valtimo/portal/gzac/objectsapi/client/ObjectsApiClient.kt]: @ivo-ritense I don't think this is applicable anymore. If it is, could you elaborate?

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**